Course_Part_10/src/TicTacToeBasic.java

```diff
@@ -0,0 +1,247 @@
+
+    public static void main(String[] args) {
+
+        playOptions = new TreeMap<>() {{
+            put("A1", "  ");
+            put("A2", "  ");
+            put("A3", "  ");
+            put("B1", "  ");
+            put("B2", "  ");
+            put("B3", "  ");
+            put("C1", "  ");
+            put("C2", "  ");
+            put("C3", "  ");
+        }};
+
+        System.out.println("Welcome to Tic-Tac-Toe");
+        displayBoardStartGame();
+        ChoosePlayMode();
+        displayBoard();
+
+        for (int turn = 1; turn < 10; ) {
+
+            playerTurn();
+            displayBoard();
+            if (isWin()) break;
+            turn++;
+
+            if (isDraw(turn)) break;
+
+            computerTurn();
+            displayBoard();
+            if (isWin()) break;
+            turn++;
+
+        }
+    }
+
+    /**
+     * Let player choose one of two play modes
+     * A: will show all the made moves and all the remaining options during the game
+     * B: will only show all the made moves during the game
+     * The selected value will be stored in the variable PlayMode
+     */
+    static void ChoosePlayMode() {
+        System.out.println();
+        System.out.println("Game has two play modes\n" +
+                "A:                 B:\n" +
+                " A1 | XX | C1          | XX |    \n" +
+                "--------------     --------------\n" +
+                " A2 | B2 | XX          |    | XX \n" +
+                "--------------     --------------\n" +
+                " OO | B3 | C3       OO |    |      "
+        );
+        do {
+            System.out.print("Please choose playmode A or B:");
+            playMode = input.next();
+        } while (!playMode.equals("A") && !playMode.equals("B"));
+        System.out.println();
+        System.out.println("Game has started - Have Fun");
+
+    }
+
+    /**
+     * Player is asked to enter his move
+     * Only the remaining options are allowed
+     * This will be validated using the isMoveNotValid method
+     * The request to enter a move will stop at a valid move selection
+     */
+    static void playerTurn() {
+        String playerMove;
+        do {
+            System.out.print("Please enter your move:");
+            playerMove = input.next();
+        } while (isMoveNotValid(playerMove));
+        playOptions.put(playerMove, "XX");
+    }
+
+    /**
+     * Using the computerPickMove method a random move will be picked and displayed
+     */
+    static void computerTurn() {
+        String computerMove = computerPickMove();
+        System.out.println("I will play at " + computerMove);
+        playOptions.put(computerMove, "OO");
+    }
+
+    /**
+     * Computer will pick a random move from the remaining options
+     *
+     * @return the random move
+     */
+    static String computerPickMove() {
+        ArrayList<String> remainingOptions = remainingPlayOptions();
+        Random randomIndex = new Random();
```

Comment:
` Random randomIndex = new Random();` kun je als static final in de class declareren en initialiseren en steeds opnieuw gebruiken, to be more efficient